patches/api/0457-Add-ShulkerDuplicateEvent.patch

```diff
@@ -0,0 +1,78 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Chase Henderson <henderson.chase@gmail.com>
+Date: Fri, 5 Jan 2024 03:50:10 -0500
+Subject: [PATCH] Add ShulkerDuplicateEvent
+
+Formatting Fixes
+
+diff --git a/src/main/java/io/papermc/paper/event/entity/ShulkerDuplicateEvent.java b/src/main/java/io/papermc/paper/event/entity/ShulkerDuplicateEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..1b3c47688e0b24fcaddda7c8728adbc51958df29
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/entity/ShulkerDuplicateEvent.java
+@@ -0,0 +1,65 @@
++package io.papermc.paper.event.entity;
++
++import org.bukkit.entity.Entity;
++import org.bukkit.entity.Shulker;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.entity.EntityEvent;
++import org.jetbrains.annotations.NotNull;
++
++/**
++ * Is called when a {@link org.bukkit.entity.Shulker} duplicates.
++ */
++public class ShulkerDuplicateEvent extends EntityEvent implements Cancellable {
++
++    private static final HandlerList HANDLER_LIST = new HandlerList();
++
++    private final Entity parent;
++    private boolean cancelled;
++
++    public ShulkerDuplicateEvent(@NotNull Entity child, @NotNull Entity parent) {
```

Comment:
seems strange to accept entity here when you always cast anyways

Reply:
I agree with mini, just make it accept Shulker